apps/dcos_dns/src/dcos_dns_handler.erl

```diff
@@ -315,6 +333,12 @@ upstream_failures(Upstream) ->
             Failures
     end.
 
+-spec(upstream_to_binary(upstream()) -> binary()).
```

Comment:
The other examples are similar, but they actualy do `a name concat with port`, and not ip. So they are simpler. I decided to live this as is. Will though write at least a test.